src/twisted/application/app.py

```diff
@@ -501,8 +501,9 @@ def opt_help_reactors(self):
         """
         rcts = sorted(self._getReactorTypes(), key=attrgetter('shortName'))
         for r in rcts:
-            self.messageOutput.write('    %-4s\t%s\n' %
-                                     (r.shortName, r.description))
+            if requireModule(r.moduleName) is not None:
+                self.messageOutput.write('    %-4s\t%s\n' %
+                                         (r.shortName, r.description))
         raise SystemExit(0)
```

Comment:
Cool.  I like the result behavior a lot.  On trunk@HEAD on Linux I get this behavior:

```
$ twistd --help-reactors
    cf          CoreFoundation integration reactor.
    default     A reasonable default: poll(2) if available, otherwise select(2).
    epoll       epoll(4)-based reactor.
    gi          GObject Introspection integration reactor.
    glib2       GLib2 event-loop integration reactor.
    gtk2        Gtk2 integration reactor.
    gtk3        Gtk3 integration reactor.
    iocp        Win32 IO Completion Ports-based reactor.
    kqueue      kqueue(2)-based reactor.
    poll        poll(2)-based reactor.
    select      select(2)-based reactor.
    win32       Win32 WaitForMultipleObjects-based reactor.
    wx          wxPython integration reactor.
```

yet I can't use many of these reactors, eg

```
$ twistd --reactor wx -n web 
...
twistd: The specified reactor cannot be used, failed with error: No module named wxPython.wx.
```

So this improves the behavior on all platforms, not just Windows.

I think that what would be one step better, and perhaps this relates to what @glyph was asking for, is to have some explanation as to why each other reactor is unavailable.  However, I'm not sure I really like my only idea for presenting this, which would be something along the lines of:

```
$ trial --help-reactors
    default     A reasonable default: poll(2) if available, otherwise select(2).
    epoll       epoll(4)-based reactor.
    poll        poll(2)-based reactor.
    select      select(2)-based reactor.
    ...
   !wx           wxPython integration reactor. (No module named wxPython.wx.)
```

Note how this identifies that the reactor exists but is not usable and gives at least a minimal description of why it is unusable.  The `!` is a bit opaque and squashing the failure into just the exception message often loses more information than you want... But perhaps we can come up with something else that accomplishes the two things this example tries to accomplish.

I'm also not sure that such an improvement needs to block _this_ work.  It's hard to say whether this is a strict improvement over trunk or a bit of sideways movement.  I _do_ like it but that's essentially up to my subjective sense of aesthetics.